src/tools/perf/lib/uct_tests.cc

```diff
@@ -463,22 +437,19 @@ class uct_perf_test_runner {
                 send_sn     = 0; /* Remote buffer will remain 0 throughout the test */
             }
 
-            set_sn(m_perf.allocator, buffer,
-                   m_perf.uct.send_mem.mem_type,
+            set_sn(buffer, m_perf.uct.send_mem.mem_type,
                    &send_sn, UCS_MEMORY_TYPE_HOST);
 
             UCX_PERF_TEST_FOREACH(&m_perf) {
                 if (flow_control) {
                     /* Wait until getting ACK from responder */
-                    sn = get_sn(m_perf.allocator,
-                                const_cast<const psn_t*>(recv_sn));
+                    sn = get_sn(const_cast<const psn_t*>(recv_sn));
```

Comment:
can we replace const_cast by reinterpret_cast?
const_cast is for stripping a const qualifier. this is not the case here
also get_sn/set_sn are convenience functions, they might as well take const void * / void * argument and cast inside the function

Reply:
> also get_sn/set_sn are convenience functions, they might as well take const void * / void * argument and cast inside the function

good catch - done

> can we replace const_cast by reinterpret_cast?
> const_cast is for stripping a const qualifier. this is not the case here

According to https://en.cppreference.com/w/cpp/language/const_cast, `const_cast` can be used to remove volatility:
"In particular, only const_cast may be used to cast away (remove) constness or volatility."